packages/flutter/lib/src/material/material.dart

```diff
@@ -228,20 +228,32 @@ class _MaterialState extends State<Material> with TickerProviderStateMixin {
       )
     );
     if (config.type == MaterialType.circle) {
-      contents = new ClipOval(child: contents);
-    } else if (kMaterialEdges[config.type] != null) {
+      contents = new PhysicalModel(
+        shape: PhysicalModelShape.oval,
+        elevation: config.elevation,
+        color: backgroundColor,
+        child: contents,
+      );
+    } else if (config.type == MaterialType.transparency) {
       contents = new ClipRRect(
-        borderRadius: radius,
+        borderRadius: radius ?? BorderRadius.zero,
         child: contents
       );
```

Comment:
We should use a ClipRect if `radius` is null.  A `ClipRect` is much less expensive because we don't need to create a temporary layer.